shopinvader_api_customer/tests/test_shopinvader_customer_api.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import json
+
+from fastapi import status
+from requests import Response
+
+from odoo.tests.common import tagged
+
+from odoo.addons.extendable_fastapi.tests.common import FastAPITransactionCase
+
+from ..routers.customer import customer_router
+
+
+@tagged("post_install", "-at_install")
+class TestShopinvaderCustomerApi(FastAPITransactionCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.Blacklist = cls.env["mail.blacklist"]
+
+        cls.pricelist = cls.env.ref("product.list0")
+        cls.user = cls.env["res.users"].create(
+            {
+                "name": "Test User",
+                "login": "test_user",
+                "email": "test@email",
+            }
+        )
+        cls.test_partner = cls.user.partner_id
+        cls.test_partner.write(
+            {
+                "phone": "12345",
+                "property_product_pricelist": cls.pricelist.id,
+            }
+        )
+        cls.default_fastapi_authenticated_partner = cls.test_partner
+        cls.default_fastapi_router = customer_router
+
+    def _call_test_client(self, url, **kwargs):
+        method = kwargs.pop("method", "get")
+        with self._create_test_client(
+            router=self.default_fastapi_router
```

Comment:
The `default_fastapi_router` is used by default by the method `_create_test_client`. You don't need to pass it to the method.